hazelcast/src/main/java/com/hazelcast/nio/Bits.java

```diff
@@ -69,265 +70,124 @@
 
     /**
      * A reusable instance of the UTF-8 charset
-     * */
+     */
     public static final Charset UTF_8 = Charset.forName("UTF-8");
     /**
      * A reusable instance of the ISO Latin-1 charset
-     * */
+     */
     public static final Charset ISO_8859_1 = Charset.forName("ISO-8859-1");
 
     private Bits() {
     }
 
     public static char readChar(byte[] buffer, int pos, boolean bigEndian) {
-        return bigEndian ? readCharB(buffer, pos) : readCharL(buffer, pos);
+        return DirectMemoryBits.readChar(buffer, pos, bigEndian);
```

Comment:
As I commented above, `Bits` should use `ByteBufferMemoryAccessor` directly here by calling `DirectMemoryBits .readChar(BYTE_BUFFER_MEMORY_ACCESSOR, buffer, pos, bigEndian)` and `DirectMemoryBits` should use `AlignedMemoryAccessor` by default.